charts/operator-wandb/templates/_redis.tpl

```diff
@@ -1,33 +1,58 @@
 {{/*
-Return name of secret where redis information is stored
+Return name of secret where redis information is stored or fallback if not present
 */}}
 {{- define "wandb.redis.passwordSecret" -}}
-{{- print .Release.Name "-redis" -}}
+{{- if .Values.global.redis.secretName -}}
+  {{ .Values.global.redis.secretName }}
+{{- else -}}
+  {{- print .Release.Name "-redis" -}}
 {{- end -}}
+{{- end }}
 
 {{/*
 Return the redis port
 */}}
 {{- define "wandb.redis.port" -}}
-{{- print $.Values.global.redis.port -}}
+{{- $redisPort := .Values.global.redis.port -}}
+{{- if .Values.global.redis.secretName -}}
+  {{- $secret := lookup "v1" "Secret" .Release.Namespace .Values.global.redis.secretName -}}
```

Comment:
I dont think the port or host need to go in the secret, they should probably just be passed as normal values